ElastixConfig.cmake.in

```diff
@@ -28,7 +28,8 @@ endif()
 
 # Import ELASTIX targets.
 set( ELASTIX_CONFIG_TARGETS_FILE "${_ELASTIXConfig_DIR}/ElastixTargets.cmake")
-if(NOT ELASTIX_TARGETS_IMPORTED)
+list( GET ELASTIX_LIBRARIES 0 _first_library)
+if(NOT ELASTIX_TARGETS_IMPORTED AND NOT TARGET ${_first_library})
```

Comment:
Can you please explain? So for SimpleITK, `TARGET ${_first_library}` is false, whereas  for ITKElastix, it is true?

Could it also be solved by simply testing if the target file exists at all? I mean:

     if((NOT ELASTIX_TARGETS_IMPORTED) AND (EXISTS ${ELASTIX_CONFIG_TARGETS_FILE}"))

Just wondering...  😃

Reply:
>  So for SimpleITK, TARGET ${_first_library} is false, whereas for ITKElastix, it is true?

Correct.

In ITKElastix, we are creating the targets:

https://github.com/InsightSoftwareConsortium/ITKElastix/blob/05cde7d3f147342702ffdb2dd6e7cbbad7e92123/CMakeLists.txt#L58

In SimpleITK, we are referencing the pre-built target in an exported targets file.

> Could it also be solved by simply testing if the target file exists at all?

`TARGET ${_first_library}` is not a file, it is the CMake binary target, i.e. what is created by `add_library`.

Reply:
Thanks Matt. I just checked your fix locally on my PC. It works well with SimpleITK (@blowekamp), _and_ it does solve the ElastixConfig.cmake error that you mentioned at https://github.com/InsightSoftwareConsortium/ITKElastix/pull/164#issuecomment-1243993000 Cool! Still some compilation errors from ITKElastix, saying 

>   ElastixTestDriver.cxx
  itkElastixRegistrationMethodTest.cxx
  F:\X\Bin\vc2019-64\ITKElastix8\_deps\elx-src\Core\Main\itkTransformixFilter.h(39,10): fatal error C1083: Cannot open include file: 'itkMesh.h': No such file or directory
Done building target "ClCompile" in project "ElastixTestDriver.vcxproj" -- FAILED.

But you already mentioned at https://github.com/InsightSoftwareConsortium/ITKElastix/pull/164#issuecomment-1281440226 that _after that patch or one like it, we will need to add `ITKMesh`, `ITKRegistrationCommon` to the `DEPENDS` in *itk-module.cmake* here_. OK 👍 !

Instead of retrieving the first item from ELASTIX_LIBRARIES, and placing it into an extra global variable `_first_library`, why not just use `elastix_lib` directly? ELASTIX_LIBRARIES is specified as follows, at https://github.com/SuperElastix/elastix/blob/0c9aa979948a5ce1226c7f18e3eb94c6cb9cf69f/Core/CMakeLists.txt#L236

    set(ELASTIX_LIBRARIES elastix_lib transformix_lib PARENT_SCOPE)

Anyway, for the sake of time, I'll just merge now!

Reply:
Thank you, @N-Dekker !

> Instead of retrieving the first item from ELASTIX_LIBRARIES, and placing it into an extra global variable _first_library, why not just use elastix_lib directly? 

This was to allow you to change the name of the target without breaking the Config file.

Reply:
@thewtex Thanks for your explanation Matt . We have no plans to change the name of the target, `elastix_lib`. I think "elastix_lib" is fine. But would you still want to change it?